### PR TITLE
[dataframe] Adding new library

### PR DESCRIPTION
**Pull request by remz1337**
This is a PR to add the [DataFrame](https://github.com/hosseinmoein/DataFrame) library.

I have followed the examples to create this port, but I still can't get it build successfully with vcpkg, so I would appreciate some support. The library compiles fine on both Linux and Windows when done manually.

Thank you

### CODE REVIEW

File: ports/dataframe/CONTROL [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Source: dataframe
+Version:  1.10.0
```
Comment (LilyWangL) [inline ports/dataframe/CONTROL:2]: Can you please add the Homepage for this port in the CONTROL file?
```
Homepage: https://github.com/hosseinmoein/DataFrame
```

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/dataframe/CONTROL (hunk 1 above)
Comment (remz1337) [reply ports/dataframe/CONTROL:2]: Done! Thank you

**Comment by LilyWangL** [comment]
@remz1337 Thanks for your PR. Now `dataframe` build failed on CI, you can click this URL and download the error log: https://dev.azure.com/vcpkg/public/_build/results?buildId=40160&view=artifacts&type=publishedArtifacts

**Comment by remz1337** [comment]
@LilyWangL I get the same logs when I try to build locally with vcpkg. As I described earlier, I don't understand why it's failling with vcpkg since I can build the library with standard cmake commands. I'm seeing some compiler warnings, but I don't understand why the standard cmake build still gets through while building with vcpkg fails.

I created this draft PR to get some help to resolve this.

Thanks

**Comment by LilyWangL** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**